### ISSUE TITLE
Localization of Strokes

### ISSUE THREAD

**Issue by jonasswiatek** [jonasswiatek/strokes#2]
Our Russian friend Vladimir wants to localize strokes to Russian (and I suppose other people might wanna do the same in their respective language).

What is the best approeach to this? currently almost all text is from the AchievementDescriptionAttribute.

**Comment by clausjoergensen** [comment]
For the GUI parts, a lot of the GUI is already localized using a Resource (RESX) file, as found here: https://github.com/jonasswiatek/strokes/tree/master/source/Strokes.GUI/Properties

To add more languages, we just need multiple resource files, such as Resources.ru-RU.resx for Russian, or Resources.da-DK.resx for Danish.  

There's a few strings that can't be localized for some reason (caused a crash-bug), postponed for later.

As for AchievementDescriptionAttribute, I'll look into designing a way to localize it.

**Comment by jonasswiatek** [comment]
We can re-engineer the AchievementDescriptionAttribute in whatever fashion we can think of if that makes localisation simpler. Currently it's not heavily integrated everywhere so it's fairly trivial

**Comment by clausjoergensen** [comment]
The ability to start localizing the project is now available. But there's a lot of achievements to modify into the @localized syntax, before we can start translating.

See [Localizing Achievements](https://github.com/jonasswiatek/strokes/wiki/Localizing-Achievements)

**Comment by jonasswiatek** [comment]
Super awesome!

**Comment by clausjoergensen** [comment]
Finished moving all hardcoded strings into resource files. Localization can now begin.

**Comment by jonasswiatek** [comment]
Bang up job!

Right now the achievement locatizations are in the Strokes.Core project - is it possible that these can reside in the BasicAchievements-project?

**Comment by clausjoergensen** [comment]
If we don't plan to have more achievement projects than the current one, it shouldn't be a problem. The problem is they need to be available from the AchievementDescriptionAttribute class, which is located in the Strokes.Core project.

**Comment by jonasswiatek** [comment]
Well, I don't think we'll have several in our distribution, however it is designed so other developers can create a VSIX that contains only Achievements (by only referencing the Strokes.Core DLL).

There is some code in Strokes.VSX that calls out for other compatible extensions and loads their assemblies into memory

**Comment by clausjoergensen** [comment]
Then we would need to load the resource files dynamically. Hmm.

**Comment by jonasswiatek** [comment]
I'll look into loading the resource assemblies dynamically.

**Comment by clausjoergensen** [comment]
Too slow!

**Comment by jonasswiatek** [comment]
Damn it! I should install Windows on the laptop I have at my TV so I can beat you to the punch!

**Comment by timdams** [comment]
Hey, here I am (from the trenches of parenthood :) )

In WPF there are some nifty XAML-based methods of doing localization. However:
a) guess that would mean some drastic refactoring? 
b) never done it before, but would be fun to try (although my Russian is a bit weak lately :p)

Perhaps this the way to go?? http://stackoverflow.com/questions/2123476/localized-attribute-parameters-in-c 

-----Oorspronkelijk bericht-----
Van: Jonas Swiatek [mailto:reply@reply.github.com] 
Verzonden: zaterdag 1 oktober 2011 18:50
Aan: Tim Dams
Onderwerp: [strokes] Localization of Strokes (#2)

Our Russian friend Vladimir wants to localize strokes to Russian (and I suppose other people might wanna do the same in their respective language).

What is the best approeach to this? currently almost all text is from the AchievementDescriptionAttribute.

## 

Reply to this email directly or view it on GitHub:
https://github.com/jonasswiatek/strokes/issues/2

**Comment by clausjoergensen** [comment]
a) No, already done that :-)
b) We use resource files.

And yes, I was inspired by the something like that. I picked a syntax with a @ denoting the localization, so we still can have un-localized stuff (mostly for legacy support)

And actually the localization here was mostly about the achievements localizations (the attribute meta-data). The WPF localization was prepared by me some months ago, since it's so easy to use databinding along with resource files.
